examples/text_classification/tc_mnli_mtdnn.ipynb

```diff
@@ -0,0 +1,1189 @@
+    "from mtdnn.configuration_mtdnn import MTDNNConfig\n",
+    "from mtdnn.data_builder_mtdnn import MTDNNDataBuilder\n",
+    "from mtdnn.modeling_mtdnn import MTDNNModel\n",
+    "from mtdnn.process_mtdnn import MTDNNDataProcess\n",
+    "from mtdnn.tasks.config import MTDNNTaskDefs\n",
+    "from mtdnn.tokenizer_mtdnn import MTDNNTokenizer\n",
+    "from utils_nlp.dataset.multinli import download_tsv_files_and_extract"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Define Configuration, Tasks and Model Objects"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define Configuration, Tasks and Model Objects\n",
+    "ROOT_DIR = TemporaryDirectory().name\n",
+    "OUTPUT_DIR = os.path.join(ROOT_DIR, 'checkpoint')\n",
+    "os.makedirs(OUTPUT_DIR) if not os.path.exists(OUTPUT_DIR) else OUTPUT_DIR\n",
+    "\n",
+    "LOG_DIR = os.path.join(ROOT_DIR, 'tensorboard_logdir')\n",
+    "os.makedirs(LOG_DIR) if not os.path.exists(LOG_DIR) else LOG_DIR\n",
+    "\n",
+    "DATA_DIR = os.path.join(ROOT_DIR, 'data')\n",
+    "os.makedirs(DATA_DIR) if not os.path.exists(DATA_DIR) else DATA_DIR\n",
+    "\n",
+    "DATA_SOURCE_DIR = os.path.join(DATA_DIR, \"MNLI\")\n",
+    "\n",
+    "# Training parameters\n",
+    "BATCH_SIZE = 16\n",
+    "MULTI_GPU_ON = True\n",
+    "MAX_SEQ_LEN = 128\n",
+    "NUM_EPOCHS = 5"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Exploring the location for our data to be downloaded, model to be checkpointed and logs to be dumped"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 19,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "/tmp/tmpd9ok4aeo/data\n",
+      "/tmp/tmpd9ok4aeo/checkpoint\n",
+      "/tmp/tmpd9ok4aeo/tensorboard_logdir\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(DATA_DIR)\n",
+    "print(OUTPUT_DIR)\n",
+    "print(LOG_DIR)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Read Dataset\n",
+    "We start by loading a subset of the data. The following function also downloads and extracts the files, if they don't exist in the data folder.\n",
+    "\n",
+    "The MultiNLI dataset is mainly used for natural language inference (NLI) tasks, where the inputs are sentence pairs and the labels are entailment indicators. The sentence pairs are also classified into *genres* that allow for more coverage and better evaluation of NLI models.\n",
+    "\n",
+    "For our classification task, we use the first sentence only as the text input, and the corresponding genre as the label. We select the examples corresponding to one of the entailment labels (*neutral* in this case) to avoid duplicate rows, as the sentences are not unique, whereas the sentence pairs are."
```

Comment:
This paragraph is not needed here. This notebook uses the entailment labels and not the genres.

Reply:
Good catch